cybersecuritytools/splunk/search.py

```diff
@@ -0,0 +1,48 @@
+from time import sleep
+from typing import Any, Dict, List
+
+from splunklib import client  # type: ignore
+from splunklib.results import ResultsReader  # type: ignore
+
+from .credentials import SplunkCredentials
+
+
+class Search:
+    def __init__(self, credentials: SplunkCredentials):
+        self.credentials = credentials
+        self.client = self.__client()
+
+    def __client(self) -> Any:
```

Comment:
Why double-underscore here?

Reply:
There was `self.client` and `self.client()` so I add the `__` for name mangling. I've renamed the method to avoid this.

Reply:
good solution